packages/sirius-web/frontend/sirius-web-view-fork/src/contributions/ForkTreeItemContextMenuContribution.tsx

```diff
@@ -0,0 +1,146 @@
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import MenuItem from '@mui/material/MenuItem';
+import React, { forwardRef, Fragment, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useRepresentationMetadata } from '../representationmetadata/useRepresentationMetadata';
+import {
+  CreateForkedStudioPayload,
+  CreateProjectSuccessPayload,
+  ForkViewMenuActionStates,
+  GQLCreateForkedStudioInput,
+  GQLCreateForkedStudioMutationData,
+  GQLErrorPayload,
+} from './ForkViewMenuAction.types';
+
+const forkViewMutation = gql`
+  mutation createForkedStudio($input: CreateForkedStudioInput!) {
+    createForkedStudio(input: $input) {
+      __typename
+      ... on CreateProjectSuccessPayload {
+        project {
+          id
+        }
+      }
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+const isErrorPayload = (payload: CreateForkedStudioPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+const isSuccessPayload = (payload: CreateForkedStudioPayload): payload is CreateProjectSuccessPayload =>
+  payload.__typename === 'CreateProjectSuccessPayload';
+
+export const ForkTreeItemContextMenuContribution = forwardRef(
+  ({ editingContextId, item }: TreeItemContextMenuComponentProps, ref: React.ForwardedRef<HTMLLIElement>) => {
+    const [createProject, { data, error }] = useMutation<GQLCreateForkedStudioMutationData>(forkViewMutation);
+    const { addErrorMessage } = useMultiToast();
+    const navigate = useNavigate();
+
+    const [state, setState] = useState<ForkViewMenuActionStates>({
+      isOpen: false,
+    });
+
+    const representationIds = item.kind.startsWith('siriusComponents://representation') ? [item.id] : [];
+    const { data: representationMetaData } = useRepresentationMetadata({ editingContextId, representationIds });
+
+    const handleClickOpen = () => {
+      setState((prevState) => ({ ...prevState, isOpen: true }));
+    };
+
+    const handleClose = () => {
+      setState((prevState) => ({ ...prevState, isOpen: false }));
+    };
+
+    const forkViewMutationHandler = () => {
+      const input: GQLCreateForkedStudioInput = {
+        id: crypto.randomUUID(),
+        editingContextId,
+        representationId: item.id,
+        tableId: '',
+      };
+      createProject({ variables: { input } });
+    };
+
+    if (data && isErrorPayload(data.createForkedStudio)) {
+      addErrorMessage(data.createForkedStudio.message);
+    }
+
+    if (error) {
+      addErrorMessage('An unexpected error has occurred, please refresh the page');
+    }
```

Comment:
Do not perform side effects while rendering the component, this should be in a useEffect

Reply:
Done